tests/app/test_jinja_filters.py

```diff
@@ -442,3 +443,60 @@ def test_other_config_visibility(
 
     other = OtherConfig(Mock(), answer_schema_textfield)
     assert other.open is expected_visibility
+
+
+@patch("app.jinja_filters.flask_babel.get_locale", Mock(return_value="en_GB"))
+def test_single_calculated_summary_question():
+    rows = map_summary_item_config(
+        group={
+            "blocks": [
+                {
+                    "id": "first-number-block",
+                    "link": "/questionnaire/first-number-block/?return_to=final-summary",
+                    "question": {
+                        "id": "first-number-question",
+                        "type": "General",
+                        "title": "First Number Question Title",
+                        "answers": [
+                            {
+                                "id": "first-number-answer",
+                                "label": "First answer label",
+                                "value": 1,
+                                "type": "currency",
+                                "currency": "GBP",
+                            }
+                        ],
+                    },
+                },
+                {
+                    "id": "second-number-block",
+                    "link": "/questionnaire/second-number-block/?return_to=final-summary",
+                    "question": {
+                        "id": "second-number-question",
+                        "type": "General",
+                        "title": "Second Number Question Title",
+                        "answers": [
+                            {
+                                "id": "second-number-answer",
+                                "label": "Second answer label",
+                                "value": 1,
+                                "type": "currency",
+                                "currency": "GBP",
+                            }
+                        ],
+                    },
+                },
+            ],
+        },
+        summary_type="CalculatedSummary",
+        answers_are_editable=True,
+        no_answer_provided="No answer Provided",
+        edit_link_text="Change",
+        edit_link_aria_label="Change your answer for",
+        calculated_question={
+            "title": "Grand total of previous values",
+            "id": "calculated-summary-question",
+            "answers": [{"id": "calculated-summary-answer", "value": "£2.00"}],
+        },
+    )
+    assert len(rows) == 3
```

Comment:
I think the assert should be more specific than this. I reckon you need to build SummaryRows and SummaryRowItems of what is expected, then assert against that.

Reply:
I think this has to stay, extra rows are not added when SummaryRow class object is being created, only at `map_summary_item_config` level.

Reply:
We build SummaryRows and SummaryRowItems now in unit-test and assertion is specific so this is resolved.